### PR TITLE
Use micronaut-gradle plugin

### LINKED ISSUE #113: Use the micronaut gradle plugin
##### **Overview of the feature request**

To have minimal custom code and use best practices from latest micronaut development we should try to use the official gradle plugin. See https://github.com/micronaut-projects/micronaut-gradle-plugin

##### **Motivation for or Use Case**

Less jhipster specific code more standard micronaut code.

##### **Related issues or PR**

* https://github.com/jhipster/generator-jhipster-micronaut/issues/89
* https://github.com/jhipster/generator-jhipster-micronaut/pull/108

### PR DESCRIPTION
**Pull request by agilob**
Replaces jib generator with micronaut-gradle-plugin.

Closes #113

### CODE REVIEW

File: generators/server/templates/gradle.properties.ejs [hunk 1]
```diff
@@ -30,6 +30,7 @@ yarn_version=<%= YARN_VERSION %>
 # Dependency versions
 jhipster_dependencies_version=<%= JHIPSTER_DEPENDENCIES_VERSION %>
 micronaut_version=<%=versions.micronaut%>
+micronautVersion=<%=versions.micronaut%>
```
Comment (agilob) [inline generators/server/templates/gradle.properties.ejs:33]: Should I deduplicate it? Micronaut function expects `micronautVersion` variable, but project already has `micronaut_version`. Duplication is minor, but I also don't want to break conventions in this project.

File: generators/server/templates/gradle/docker.gradle.ejs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+dockerfile {
+    baseImage = "openjdk:14-alpine"
```
Comment (agilob) [inline generators/server/templates/gradle/docker.gradle.ejs:2]: `baseImage` is using by default "openjdk:14-alpine", added here only as "documentation", so users can switch without digging into docs. Same for nativeImage below. Not tested yet, but if anyone isn't using metrics and doesn't have frontend, it should work(maybe).

**Comment by agilob** [comment]
I've built is using docker-compose:

```
services:
  mhip:
    image: mhip2:0.0.1-SNAPSHOT
    container_name: mhip
    links:
      - agilob-postgresql
    networks:
      - agilob-network
    environment:
      - _JAVA_OPTIONS=-Xmx512m -Xms256m
      - MICRONAUT_ENVIRONMENTS=prod
    ports:
      - 8080:8080
```

You can see java_options being picked up from env and used :)

Screenshot_20201003_161215

File: generators/server/templates/gradle.properties.ejs (hunk 1 above)
Comment (atomfrede) [reply generators/server/templates/gradle.properties.ejs:33]: Let's de-duplicate it I would say.

File: generators/common/templates/README.md.ejs [hunk 3]
```diff
@@ -281,8 +281,12 @@ To stop it and remove the container, run:
 You can also fully dockerize your application and all the services that it depends on.
 To achieve this, first build a docker image of your app by running:
 
-    <% if (buildTool === 'maven') { %>./mvnw -Pprod verify jib:dockerBuild<% } %><% if (buildTool === 'gradle') { %>./gradlew bootJar -Pprod jibDockerBuild<% } %>
-
+    <% if (buildTool === 'maven') { %>./mvnw -Pprod verify jib:dockerBuild<% } %><% if (buildTool === 'gradle') { %>../gradlew clean assemble dockerBuild -Pprod<% } %>
+    <% if (buildTool === 'gradle') { %>
+    Gradle uses official [micronaut-gradle-plugin](https://github.com/micronaut-projects/micronaut-gradle-plugin) for Docker image generation.
+    <% } %><% if (buildTool === 'gradle') { %>
```
Comment (atomfrede) [inline generators/common/templates/README.md.ejs:287]: Looks like a mistake here.

File: generators/server/templates/build.gradle.ejs [hunk 4]
```diff
@@ -52,6 +49,15 @@ plugins {
     //jhipster-needle-gradle-plugins - JHipster will add additional gradle plugins here
 }
 
+micronaut {
+    runtime "netty"
+    testRuntime "junit5"
+    processing {
+        incremental true
+        annotations "example.mn.*"
```
Comment (atomfrede) [inline generators/server/templates/build.gradle.ejs:57]: This should be the selected packageName I think.

**Review by atomfrede — CHANGES_REQUESTED** [review]
Nice work @agilob :+1:  Two minor remarks

**Review by atomfrede — APPROVED** [review]
Don't forget to claim the bounty @agilob

**Comment by agilob** [comment]
Claimed https://opencollective.com/generator-jhipster/expenses/26007

**Pull request merged**